code/modules/mob/living/carbon/xenomorph/abilities/ability_helper_procs.dm

```diff
@@ -286,27 +286,26 @@
 	var/turf/prev_turf = loc
 
 	var/distance = 0
-	for(var/turf/T in turflist)
+	for(var/turf/turf in turflist)
 		distance++
 
 		if(!prev_turf && length(turflist) > 1)
 			prev_turf = get_turf(src)
 			continue //So we don't burn the tile we be standin on
 
-		if(T.density || istype(T, /turf/open/space))
+		if(turf.density || istype(turf, /turf/open/space))
 			break
 		if(distance > distance_max)
 			break
-
-		var/atom/movable/temp = new spray_path()
-		var/atom/movable/AM = LinkBlocked(temp, prev_turf, T)
+		var/obj/effect/xenomorph/spray/temp = new spray_path()
```

Comment:
Spray path can potentially not be `/obj/effect/xenomorph/spray` Thats just what it is by default.